src/miniscript/context.rs

```diff
@@ -384,12 +391,17 @@ impl ScriptContext for Legacy {
         }
 
         match ms.node {
-            Terminal::PkK(ref key) if key.is_x_only_key() => {
-                return Err(ScriptContextError::XOnlyKeysNotAllowed(
-                    key.to_string(),
-                    Self::name_str(),
-                ))
-            }
+            Terminal::PkK(ref key) => match key {
+                KeyExpr::<Pk>::SingleKey(pk) => {
+                    if pk.is_x_only_key() {
+                        return Err(ScriptContextError::XOnlyKeysNotAllowed(
+                            pk.to_string(),
+                            Self::name_str(),
+                        ));
+                    }
+                }
+                KeyExpr::<Pk>::MuSig(_) => return Err(ScriptContextError::MusigNotAllowed),
```

Comment:
Add a string parameter as `MusigNotAllowed(String)` and pass Self::name_str. This way whenever user can also see the error context in which musig was not allowed.

Reply:
Done. added a string parameter.